src/tools/idlpp/src/main.c

```diff
@@ -965,18 +965,21 @@ static void put_a_line(
  */
 {
     size_t  len;
-    char *  out_p;
-    char *  tp;
 
     if (no_output)
         return;
     len = strlen( out);
-    tp = out_p = out + len - 2;             /* Just before '\n'     */
-    while (char_type[ *out_p & UCHARMAX] & SPA)
-        out_p--;                    /* Remove trailing white spaces */
-    if (out_p < tp) {
-        *++out_p = '\n';
-        *++out_p = EOS;
+    if (len > 2)
```

Comment:
A line with just a whitespace char and a line-break is not trimmed (or is that intentional?). I think this could be `>= 2` and ..

Reply:
It is intentional, but it is good to double-check. From my reading of the code, the intent is that there is always at least a character, followed by a newline, followed by a terminating 0. This then makes the `-2` valid. The loop implies that the character is assumed to be a "non-whitespace" one.

Firstly, this means that under the original assumptions, whether or not you actually do anything in the case of `len == 2` is irrelevant.

The bug report only proves that the "non-whitespace" assumption is false. It doesn't prove that the character+`\n`+`\0` assumption is false. So my decision to add a test for `len > 2` (or, equivalently under the original assumption, `len >= 2`) is simply because it wasn't immediately obvious that `len < 2` is truly impossible.

So, the only surprising bit is that in the case of an input of the form: whitespace character+`\n`+`\0`, it outputs the whitespace character. Considering that this is merely the preprocessor and the IDL lexer doesn't mind the white space, I figured this edge case didn't really matter.